cognite/client/data_classes/simulators/models.py

```diff
@@ -41,8 +45,22 @@ def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None =
             model_external_id=resource["modelExternalId"],
             file_id=resource["fileId"],
             description=resource.get("description"),
+            external_dependencies=SimulatorModelRevisionExternalDependency._load_list(
+                resource["externalDependencies"], cognite_client
+            )
+            if "externalDependencies" in resource
+            else None,
         )
 
+    def dump(self, camel_case: bool = True) -> dict[str, Any]:
+        output = super().dump(camel_case=camel_case)
+        if isinstance(self.external_dependencies, list) and all(
+            isinstance(item, SimulatorModelRevisionExternalDependency) for item in self.external_dependencies
+        ):
+            output["externalDependencies"] = [item.dump(camel_case=camel_case) for item in self.external_dependencies]
```

Comment:
Typing already ensures that this is `list[SimulatorModelRevisionExternalDependency] | None`
```suggestion
        if self.external_dependencies is not None:
            output["externalDependencies"] = [item.dump(camel_case=camel_case) for item in self.external_dependencies]
```